platform/mbed_mpu_mgmt.c

```diff
@@ -17,19 +17,19 @@
 #include "platform/mbed_mpu_mgmt.h"
 #include "platform/mbed_critical.h"
 #include "platform/mbed_error.h"
+#include "platform/mbed_assert.h"
 #include "hal/mpu_api.h"
 #include <limits.h>
 
+#if DEVICE_MPU && !MBED_CONF_PLATFORM_DISABLE_MPU
+
 static uint16_t mem_xn_lock;
 static uint16_t mem_wn_lock;
 
 void mbed_mpu_manager_lock_ram_execution()
```

Comment:
@deepikabhavnani initially I wrapped ScopedRamExecutionLock and ScopedRomWriteLock in `#if DEVICE_MPU && !MBED_CONF_PLATFORM_DISABLE_MPU`  but I found it didn't actually save any space, as the compiler was already removing it. I was using a GCC_ARM development build.

Reply:
Debug builds would be forced to keep this around though, so that may be a good reason to remove it. Same would go for the RAII class which adds more code at least in debug builds. @kjbracey-arm @donatieng @bulislaw @deepikabhavnani what are your thoughts on this?

Reply:
If we need RAII class for users we can let it be in code space, but internally in platform and drivers we can use internal functions instead of class.. can check how much space is saved and is worth or not ?

Reply:
Linker should drop the class and size if not used in debug builds as well

Reply:
I'm not too bothered about debug builds - lots of good C++ practice causes bloat if optimisation is fully disabled. You can't start trying to avoid typical teeny inline C++ methods just to "optimise" debug builds - which is something of an oxymoron.

If anything we need more use of scoped locks than manual locks - there's a continuous low-level flow of "add missing unlock" PRs.